pkg/cmd/rollbackcopy/tarutils.go

```diff
@@ -0,0 +1,90 @@
+package rollbackcopy
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func createTarball(tarballFilePath string, filePaths []string, prefixTrim string) error {
+	file, err := os.Create(tarballFilePath)
```

Comment:
Security issue: I am able as nonroot user get access to all TLS keys for `etcd` , `kube-apiserver`, `kube-scheduler`  and `kube-controller-manager`

```
[core@ip-10-0-162-37 ~]$ tar -xvf /etc/kubernetes/rollbackcopy/currentVersion.latest/static_kuberesources_2020-06-16_013447.tar.gz 
static-pod-resources/kube-apiserver-pod-6/configmaps/bound-sa-token-signing-certs/service-account-001.pub
static-pod-resources/kube-apiserver-pod-6/configmaps/config/config.yaml
static-pod-resources/kube-apiserver-pod-6/configmaps/etcd-serving-ca/ca-bundle.crt
[..]
```
```
[core@ip-10-0-162-37 ~]$ cat static-pod-resources/etcd-pod-3/secrets/etcd-all-serving/etcd-serving-ip-10-0-162-37.us-west-1.compute.internal.key
-----BEGIN RSA PRIVATE KEY-----
MIIEowIBAAKCAQEAzMp+z9RuvEFWA78p8FnFiazRakNCJLkeZtuWkCbSbYdWm/7N
vWRNuBfbdgnckXHwxfNacCDtqRYNxnjQ6ZIeg5mukQGFB3C2//J8LXaPiWAbUQ84
xvzmpddmJrSNOLAZ2nYXkPqukCPqIAEm5CeJYh17rNhi0kJuwsSGpDzXtKfusO6n
mOPEoYLjlcbvPqQWMbcvgMUziU4YjpYYyXW2d2QlKLD7x8ppde+y2NJoW/C4vbqB
CMUSMm2mstK/dgrZTvBemBYgTLQrf39u9D+bSyJH7tbH0t8rgFY9S83BLYgVoduN
EKlH0ExQ3a0wVSvW8DY4ZDJcPedo+oMrgNfn6wIDAQABAoIBAGADWC0MRdaRM0iS
aZE8JXcExTHy0qHxAKXsscxnxyR+3UZJ0h/HVQDIfqX6vaSbLO1ByNblOUebBV5V
6mx3i0Lb3JIzijEQ0VtyTnHqMiWQH36DinUQwPpj4Iqo/8TiG1znHOll/k2p6OyU
Xa1+M9UCSaeh08MMTm2WQiHiPKd/dSPhfb4qMc8ZKx0te8aJy+0i2HPcQUwwHtJi
fZt59jUAiFhFD+QWMBq316cMlwbc5OcoFAYPCSNyoFKtG6a5aPTdAspkM4YH3fSB
oR6xOniEc7HNhDiPwtuKpTiiB2XvgoIXv1WynPS25YB8Y7XBw9bzs82LIm99yNWe
m7p8+4ECgYEA1hekI/X+Yfqm0U9fggnpJ0fiHq5jL0ljnZZovRIPvQAa+yZfkJDp
m0TZLVSs3AEXUXgaAXPneXu06fmW4dGRvtVVwPEZBTiulwDmVLYgkQheTx70i4rs
CpWZtYfaiu0FO3l3ovW5CFA2OJA7/L94UCwtc3kglABhQhFA36hQ+yECgYEA9ODB
U9T0FSw9+9weqos3hY8+WTwwjMk17IJ9PiOdN2EROv/BvtrpgIyjH9QhCCbktAGu
mjmkKhkWdEKcnF+PnkwHoXkjrqoDpbAkT9AW/6X4xslO6f8Y8VUAz4YrGcqLFIhz
5kK251+G+Jjg3IRotgisttgMd3v1wsTe54oi7YsCgYBv7nxKsNoGr913oV/1xOi5
OMPn9pS0rFIAmhq4fzDrR5rTpARiuODp5N0G7jZR7+AZIAm4qLhks2Cq02snNUJV
24OyFFtYd4IQQNArie0M5qsoAq8oJcGy5vQuDJEeAT2zEpaSBJZw2j7+46n5QsHM
aMx6iuGVf31Z5Vtdk8qDAQKBgQCHQcmnK13yf/BwuCG8I490Vy2iDmvmp5EAZQcW
lodVOmlRBEmYyMPm/Edx4jKolQsIHlG54BK8KgVMDCsC+ImUbvxcBVRfOosvLn6w
EwlRBfTGoBTVjNDTJBeBiOMvqL9vhPs3eSwb2pjroU26n+xSd7ImgGlKag7SGPni
PFuUhwKBgEjLvtuha4gY4TjCdm5R2UMAqtb0sbbO1pPcdM1/SvffgIXaXaXBY83U
foKG8RHl9QDZ5FhitWCZj10aD491IawkdLq6lRyXTZU1pEBPX6WLQDj+PpXo5dPS
Pu70FPJz9u/w5XAsWb3pKYYmn6cpxFla+shZFjy6fI2LTn2qC5AA
-----END RSA PRIVATE KEY-----
```
This tar needs to be 600

Reply:
```
[core@ip-10-0-179-219 ~]$ ls -la /etc/kubernetes/rollbackcopy/currentVersion.prev
total 41532
drwxr-xr-x. 2 root root      118 Jun 16 01:19 .
drwxr-xr-x. 4 root root       62 Jun 16 01:29 ..
-rw-r--r--. 1 root root       76 Jun 16 00:55 backupenv.json
-rw-------. 1 root root 42467360 Jun 16 00:55 snapshot_2020-06-16_005505.db
-rw-r--r--. 1 root root    51585 Jun 16 00:55 static_kuberesources_2020-06-16_005505.tar.gz
```

Reply:
added tar to be 600.